### PR TITLE
fix url highlighting due to deprecation of vte_terminal_match_add_gregex

### PR DESCRIPTION
**Pull request by norbusan**
Fixes Issue 329

### CODE REVIEW

**Comment by sc0w** [comment]
I see in `configure.ac` `VTE_REQUIRED=0.48`

I think you don't need the checks

**Comment by norbusan** [comment]
Oh, so we could drop support for the old way completely, that would simplify the patch.

Should I update it?

**Comment by sc0w** [comment]
yes, please

**Comment by norbusan** [comment]
Ok, I force pushed now with the checks removed and all switched to VteRegexp

File: src/terminal-screen.c [hunk 1]
```diff
@@ -45,6 +45,9 @@
 
 #include "eggshell.h"
 
+#define PCRE2_CODE_UNIT_WIDTH 0
```
Comment (norbusan) [inline src/terminal-screen.c]: Hontestly, I am not sure about what this does, I got it from the patch https://sourceforge.net/p/lxde/bugs/932/attachment/0001-Don-t-use-deprecated-vte_terminal_match_add_gregex.patch

**Comment by sc0w** [comment]
New warning detected in this PR:

```
terminal-screen.c: In function 'terminal_screen_class_init':
terminal-screen.c:596:15: warning: assignment to 'VteRegex **' {aka 'struct _VteRegex **'} from incompatible pointer type 'GRegex **' {aka 'struct _GRegex **'} [-Wincompatible-pointer-types]
  596 |  skey_regexes = g_new0 (GRegex*, n_skey_regexes);
      |               ^
```

**Comment by norbusan** [comment]
@sc0w Thanks, I missed that one during the simplification (VTE check removal), pushed a new commit that changes the type.

**Comment by sc0w** [comment]
now I noticed these warnings, but we have similar warnings in master

```
terminal-screen.c:186:2: warning: missing initializer for field 'flags' of 'TerminalRegexPattern' {aka 'const struct <anonymous>'} [-Wmissing-field-initializers]
  186 |  { "s/key [[:digit:]]* [-[:alnum:]]*",         FLAVOR_AS_IS },
      |  ^
terminal-screen.c:164:10: note: 'flags' declared here
  164 |  guint32 flags;
      |          ^~~~~

terminal-screen.c:187:2: warning: missing initializer for field 'flags' of 'TerminalRegexPattern' {aka 'const struct <anonymous>'} [-Wmissing-field-initializers]
  187 |  { "otp-[a-z0-9]* [[:digit:]]* [-[:alnum:]]*", FLAVOR_AS_IS },
      |  ^
terminal-screen.c:164:10: note: 'flags' declared here
  164 |  guint32 flags;
      |          ^~~~~
```

**Comment by norbusan** [comment]
@sc0w these warnings have been there since before, because the initializers for these structs always were missing the flag value. I guess putting a simple 0 at the end of all would silence the warnings.

**Comment by sc0w** [comment]
can you fix it here please?

**Comment by norbusan** [comment]
Ok, I'll add a commit fixing this. Please wait a bit.

**Comment by norbusan** [comment]
@sc0w I have pushed a trivial fix, but interestingly, *I* don't see the warning you pasted abvoe about missing initializers. Did you do something special when compiling?

Futhermore, there are more things to be fixed: The grexep stuff is still used in the search function, I will look into fixing the search, too.

**Comment by norbusan** [comment]
@sc0w I have now fixed also the search problem - at least for me searching works again. There is a strange warning message issued while searching:
```
(mate-terminal:54775): VTE-WARNING **: 07:49:15.285: (../src/vtegtk.cc:2477):void vte_terminal_search_set_regex(VteTerminal*, VteRegex*, guint32): runtime check failed: (regex == nullptr || _vte_regex_has_multiline_compile_flag(regex))
```
I checked the API, and passing `NULL` should be correct for clearing the regex, according the the vte api document. And I don't see any multline compile flag set.

Anyway, functionality-wise I can at least search with default settings, didn't try to do anything fancy, though.

**Comment by sc0w** [comment]
> Did you do something special when compiling?

I just saw the travis logs (`--enable-compile-warnings=maximum`)

please, can you squash https://github.com/mate-desktop/mate-terminal/pull/330/commits/8e434e6fd564c212e33d256c1f5b174bef5483e8 https://github.com/mate-desktop/mate-terminal/pull/330/commits/90f80adc82ce9ba15d74d01948c4ace262296954 https://github.com/mate-desktop/mate-terminal/pull/330/commits/b171a434055957716420ece6caef9a9dcca15f8f into one unique commit?, and it will be ready to go.

I think https://github.com/mate-desktop/mate-terminal/pull/330/commits/3727f7c5467e4693dcc7640d7f2381a47bb48454 is unrelated and it can be handled in new PR, please open new PR with it.

File: src/terminal-screen.c (hunk 1 above)
Comment (norbusan) [reply src/terminal-screen.c]: Understtod, necessary for loading pcre2.h, otherwise it errors out.

**Comment by norbusan** [comment]
Hi @sc0w 
Done all that. Search related PR is here https://github.com/mate-desktop/mate-terminal/pull/332

**Comment by norbusan** [comment]
Thanks for merging!
Now I am looking into all the other deprecated warnings, there are quite a lot of them ...

**Comment by raveit65** [comment]
@norbusan 
Can this be used for distros with vte-0.58.3, eg. fedora 31?
https://koji.fedoraproject.org/koji/buildinfo?buildID=1472915
My plan is to update fedora 31 with MATE 1.24 after fedora 32 is released in a few weeks.

**Comment by norbusan** [comment]
@raveit65 Yes, the functionality used is there since vte .48 afair, so if you have 0.58 then it should be no problem to use.

**Pull request merged**